modules/miner/update.go

```diff
@@ -1,11 +1,205 @@
+			// Put back in transactions removed.
+			for _, v := range bottomSets {
+				m.blockMapHeap.Push(v)
+			}
+			// Go to the next candidate set.
+			break
+		}
+		nextSet := m.blockMapHeap.Pop()
+
+		bottomSets = append(bottomSets, nextSet)
+
+		// Calculating fees to compare total fee from those sets removed
+		// and the current set s.
+		totalFeeFromNextSet := nextSet.set.averageFee.Mul64(nextSet.set.size)
+		totalBottomFees := averageFeeOfBottomSets.Mul64(sizeOfBottomSets).Add(totalFeeFromNextSet)
+		sizeOfBottomSets += nextSet.set.size
+		averageFeeOfBottomSets := totalBottomFees.Div64(sizeOfBottomSets)
+
+		// If the average fee of the bottom sets from the block is higher
+		// than the fee from this candidate set, put the candidate into the
+		// overflow MapHeap.
+		if averageFeeOfBottomSets.Cmp(candidateSet.averageFee) == 1 {
+			// candidateSet goes into the overflow.
+			m.overflowMapHeap.Push(elem)
+			// Put transaction sets from bottom back into the blockMapHeap.
+			for _, v := range bottomSets {
+				m.blockMapHeap.Push(v)
+			}
+			// Go to the next candidate set.
+			break
+		}
+
+		// Check if the candidateSet can fit in the block.
+		if m.blockMapHeap.size-sizeOfBottomSets+candidateSet.size < types.BlockSizeLimit-5e3 {
+			//Place candidate into block,
+			m.blockMapHeap.Push(elem)
+
+			// Places transactions removed from block heap into
+			// the overflow heap.
+			for _, v := range bottomSets {
+				m.overflowMapHeap.Push(v)
+			}
+			break
+		}
+
+	}
+}
+
+// deleteReverts deletes transactions from the miner's transaction selection
+// which are no longer in the transaction pool.
+func (m *Miner) deleteReverts(diff *modules.TransactionPoolDiff) {
+	// Delete the sets that are no longer useful. That means recognizing which
+	// of your splits belong to the missing sets.
+	for _, id := range diff.RevertedTransactions {
+		// Look up all of the split sets associated with the set being reverted,
+		// and delete them. Then delete the lookups from the list of full sets
+		// as well.
+		splitSetIndexes := m.fullSets[id]
+		for _, ss := range splitSetIndexes {
+			m.deleteMapElementTxns(splitSetID(ss))
+		}
+		delete(m.fullSets, id)
+	}
+}
+
+func (m *Miner) deleteMapElementTxns(id splitSetID) {
+	_, inBlockMapHeap := m.blockMapHeap.selectID[id]
+	_, inOverflowMapHeap := m.overflowMapHeap.selectID[id]
+
+	// If the transaction set is in the overflow, we can just delete it.
+	if inOverflowMapHeap {
+		m.overflowMapHeap.RemoveSetByID(id)
+		return
+	}
+
+	if inBlockMapHeap {
+		// Remove from blockMapHeap.
+		m.blockMapHeap.RemoveSetByID(id)
+
+		// Promote sets from overflow heap to block if possible.
+		for {
+			overflowElem, canPromote := m.overflowMapHeap.Peek()
+			if canPromote && m.blockMapHeap.size+overflowElem.set.size < types.BlockSizeLimit-5e3 {
+				promotedElem := m.overflowMapHeap.Pop()
+				m.blockMapHeap.Push(promotedElem)
+				continue
+			}
+			break
+		}
+	}
+	delete(m.splitSets, id)
```

Comment:
or maybe the `inOverflow/Block` blocks need to be an `if else` instead of consecutive `if`